guides/plugins/plugins/storefront/add-dynamic-content-via-ajax-calls.md

```diff
@@ -0,0 +1,138 @@
+# Add dynamic content via ajax calls
+
+This guide will show you how to add dynamic content to your storefront.
+It combines and builds upon the the guides about [adding custom Javascript](./add-custom-javascript.md) and [adding a custom controller](./add-custom-controller.md), so you should probably read them first.
+
+## Setting up the Controller
+
+For this guide we will use a very simple controller that returns a timestamp wrapped in the JSON format.
+
+As mentioned before this guide builds up upon the [adding a custom controller](./add-custom-controller.md) guide.
+This means that this article will only cover the differences between returning a template and a `JSON` response and making it accessible to `XmlHttpRequests`.
+
+```php
+<?php declare(strict_types=1);
+
+namespace SwagBasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+   /**
+    * @Route("/example", name="frontend.example.example", defaults={"XmlHttpRequest"=true}, methods={"GET"})
+    */
+    public function showExample(): JsonResponse
+    {
+        return new JsonResponse(['timestamp' => gmdate("Y-m-d\TH:i:s\Z")]);
+    }
+}
+```
```

Comment:
``````suggestion
{% code title="<plugin base>/Storefront/Controller/ExampleController.php" %}
```php
<?php declare(strict_types=1);

namespace SwagBasicExample\Storefront\Controller;

use Shopware\Core\Framework\Routing\Annotation\RouteScope;
use Shopware\Storefront\Controller\StorefrontController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\Routing\Annotation\Route;

/**
 * @RouteScope(scopes={"storefront"})
 */
class ExampleController extends StorefrontController
{
   /**
    * @Route("/example", name="frontend.example.example", defaults={"XmlHttpRequest"=true}, methods={"GET"})
    */
    public function showExample(): JsonResponse
    {
        return new JsonResponse(['timestamp' => gmdate("Y-m-d\TH:i:s\Z")]);
    }
}
```
{% endcode %}
``````

Reply:
Pretty please do this for the rest of the code blocks aswell